RecoTauTag/RecoTau/plugins/PFRecoTauDiscriminationByIsolation.cc

```diff
@@ -195,10 +228,29 @@ class PFRecoTauDiscriminationByIsolation : public PFTauDiscriminationProducerBas
   double offsetRelativeSumPt_;
   double customIsoCone_;
   
+  bool applyPhotonPtSumOutsideSignalConeCut_;
+  double maxAbsPhotonSumPt_outsideSignalCone_;
+  double maxRelPhotonSumPt_outsideSignalCone_;
+
+  bool applyFootprintCorrection_;
+  struct footprintCorrectionType
+  {
+    footprintCorrectionType(const std::string& selection, const std::string& offset)
+      : selection_(selection),
+	offset_(offset)
+    {}
+    ~footprintCorrectionType() {}
+    StringCutObjectSelector<PFTau> selection_;
+    StringObjectFunction<PFTau> offset_;
+  };
+  std::vector<footprintCorrectionType*> footprintCorrections_;
```

Comment:
this one is less trivial, std::move is needed
if I guessed the line number right
```diff
-	footprintCorrectionType* footprintCorrection = new footprintCorrectionType(selection, offset);
-	footprintCorrections_.push_back(footprintCorrection);
+      std::unique_ptr<FootprintCorrection> footprintCorrection(new FootprintCorrection(selection, offset));
+      footprintCorrections_.push_back(std::move(footprintCorrection));
```